synapse/federation/transport/server/__init__.py

```diff
@@ -324,6 +327,13 @@ def register_servlets(
             )
 
         for servletclass in DEFAULT_SERVLET_GROUPS[servlet_group]:
+            # Only allow the `/timestamp_to_event` servlet if msc3030 is enabled
+            if (
+                servletclass == FederationTimestampLookupServlet
+                and not hs.config.experimental.msc3030_enabled
+            ):
+                continue
```

Comment:
Because `FederationTimestampLookupServlet` is bundled up in the `FEDERATION_SERVLET_CLASSES` which is an immutable tuple without any access to `hs`, I had to feature flag the endpoint this way. Open to any better solutions 😀

Reply:
Perhaps we could register the servlet manually?
```py
if "federation" in servlet_groups and hs.config.experimental.msc3030_enabled:
    FederationTimestampLookupServlet(...).register(resource)
```

Reply:
That seems like a lot more boilerplate in the `...` to keep in sync with each other.

What do you prefer?

Reply:
I've got no strong preference here. Either way works for me.